nixos/modules/security/wrappers/wrapper.c

```diff
@@ -171,6 +171,7 @@ static int make_caps_ambient(const char *self_path) {
 
 int main(int argc, char **argv) {
     ASSERT(argc >= 1);
+    ASSERT(strlen(argv[0]) < 512);
```

Comment:
Please document this so later readers don't have to dig through the git history for context.

Reply:
I would like to keep this constraint even after the CVE is gone, maintaining strictness in what we let through the wrappers as a matter of general practice. If anyone wants to see specifically why and when it was added, they can get it from git blame. I'm happy to add a comment about *that* itself, though.

Reply:
Sure, I'm not suggesting that this goes away after some given point in time, I'm saying I'd like to see some explanation for it which, for example, references the CVE as a motivation of why we care about sanitizing argv0. Without context, this is a puzzling check, especially since the same sanitization is not being applied to other inputs going through wrappers (e.g. argv 1..N, env vars, possibly other things).

Reply:
Newly added comments look good, thanks!